asllib/doc/Expressions.tex

```diff
@@ -1315,30 +1368,80 @@ \subsubsection{Prose}
 All of the following apply:
 \begin{itemize}
   \item $\ve$ denotes the \arrayaccess{} expression with base $\ebase$ and index $\eindex$;
-  \item $(\tbase,\ebasep)$ is the result of annotating the expression $\ebase$ in $\tenv$\ProseOrTypeError;
-  \item $\tbase$ has the structure of an array with index $\size$ and element type $\telem$\ProseOrTypeError;
-  \item annotating the expression $\eindex$ in $\tenv$ yields $(\tindexp, \eindexp)$\ProseOrTypeError;
-  \item determining the type of the array index for $\size$ in $\tenv$ via \\ $\typeofarraylength$
-        yields $\wantedtindex$;
-  \item determining whether $\tindexp$ \typesatisfies\ $\wantedtindex$ yields \\
-        $\True$\ProseOrTypeError;
-  \item $\vt$ is $\telem$;
-  \item $\newe$ is the \arrayaccess\ expression for $\ebasep$ and index $\eindexp$, that is, $\EGetArray(\ebasep, \eindexp)$.
+  \item \Proseannotateexpr{$\tenv$}{$\ebase$}{\\ $(\tbase,\ebasep, \vsesbase)$}\ProseOrTypeError;
+  \item obtaining the \underlyingtype\ of $\tbase$ in $\tenv$ yields $\tanonbase$\ProseOrTypeError;
+  \item view $\tanonbase$ as the array type with size expression $\size$ and element type $\telem$,
```

Comment:
Fixed. By the way, you can't obtain an underlying type and return an error if it's not the type you'd like, since they `make_anonymous` function only returns a type, it doesn't perform the extra check, which you have to do yourself.